### PR TITLE
Added page on how to extend Deploy for custom data and property editors

### PR DESCRIPTION
**Pull request by AndyButland**
I've added this as had a small update to make sure we record as part of 4.1.1, but realised we didn't really have a home for it.

It's worth having available for anyone looking to extend Deploy with their own data or custom property types.  Some of this I've adapted from internal documentation, some from code samples I've written for development and testing tasks.

### CODE REVIEW

**Comment by jonat123** [comment]
@AndyButland Good PR overall, there are a few complaints from vale and I have added some suggestions to them as well 🙂

**Comment by AndyButland** [comment]
Thanks, I've updated based on the linter comments now.

**Comment by jonat123** [comment]
@AndyButland great! I'll get it merged in when the new deploy version is released tomorrow :)

**Review by jonat123 — APPROVED** [review]

File: Add-ons/Umbraco-Deploy/Extending/index.md [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+versionFrom: 8.0.0
+meta.Title: "Extending Umbraco Deploy"
+meta.Description: "How to extend Umbraco Deploy to synchronize custom data"
+---
+
+# Extending Umbraco Deploy
+
+Umbraco Deploy supports the deployment of CMS schema information, definitions from the HQ's Forms package, along with managed content and media.  In addition it can be extended by package or custom solution developers to allow the deployment of custom data, such as that stored in your own database tables.
+
+Currently supported is the ability to hook into the disk based serialization and deployment - similar to that used for Umbraco document types and data types.  In a later release we plan to also all deployment via the back-office - similar to how Umbraco content and media can be queued for transfer and restored.
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply Add-ons/Umbraco-Deploy/Extending/index.md]: Should be "...we plan to also support deployment of..." - so no, not all, rather what's appropriate (so in the 
same way that editor's deploy content and developers deploy documennt types, there can be a similar split for custom data).

**Comment by AndyButland** [comment]
Thanks - it can go whenever though actually, it's not really related to tomorrow patch release (i.e. it's documenting already existing features).

**Comment by jonat123** [comment]
@AndyButland Ahhh okay, then i will just get it merged in!

**Pull request merged**